pkg/webhook/framework.go

```diff
@@ -0,0 +1,247 @@
+		}
+		convertedCR.SetAPIVersion(convertRequest.DesiredAPIVersion)
+		convertedObjects = append(convertedObjects, runtime.RawExtension{Object: convertedCR})
+	}
+	return &v1beta1.ConversionResponse{
+		ConvertedObjects: convertedObjects,
+		Result:           statusSucceed(),
+	}
+}
+
+// doConversionV1 converts the requested objects in the v1 ConversionRequest using the given conversion function and
+// returns a conversion response. Failures are reported with the Reason in the conversion response.
+func doConversionV1(convertRequest *v1.ConversionRequest, convert convertFunc) *v1.ConversionResponse {
+	var convertedObjects []runtime.RawExtension
+	for _, obj := range convertRequest.Objects {
+		cr := unstructured.Unstructured{}
+		if err := cr.UnmarshalJSON(obj.Raw); err != nil {
+			klog.Error(err)
+			return &v1.ConversionResponse{
+				Result: metav1.Status{
+					Message: fmt.Sprintf("failed to unmarshall object (%v) with error: %v", string(obj.Raw), err),
+					Status:  metav1.StatusFailure,
+				},
+			}
+		}
+		convertedCR, status := convert(&cr, convertRequest.DesiredAPIVersion)
+		if status.Status != metav1.StatusSuccess {
+			klog.Error(status.String())
+			return &v1.ConversionResponse{
+				Result: status,
+			}
+		}
+		convertedCR.SetAPIVersion(convertRequest.DesiredAPIVersion)
+		convertedObjects = append(convertedObjects, runtime.RawExtension{Object: convertedCR})
+	}
+	return &v1.ConversionResponse{
+		ConvertedObjects: convertedObjects,
+		Result:           statusSucceed(),
+	}
+}
+
+func serve(w http.ResponseWriter, r *http.Request, convert convertFunc) {
+	var body []byte
+	if r.Body != nil {
+		if data, err := io.ReadAll(r.Body); err == nil {
+			body = data
+		}
+	}
+
+	contentType := r.Header.Get("Content-Type")
+	serializer := getInputSerializer(contentType)
+	if serializer == nil {
+		msg := fmt.Sprintf("invalid Content-Type header `%s`", contentType)
+		klog.Errorf("%s", msg)
+		http.Error(w, msg, http.StatusBadRequest)
+		return
+	}
+
+	klog.V(2).Infof("handling request: %v", body)
+	obj, gvk, err := serializer.Decode(body, nil, nil)
+	if err != nil {
+		msg := fmt.Sprintf("failed to deserialize body (%v) with error %v", string(body), err)
+		klog.Error(err)
+		http.Error(w, msg, http.StatusBadRequest)
+		return
+	}
+
+	var responseObj runtime.Object
+	switch *gvk {
+	case v1beta1.SchemeGroupVersion.WithKind("ConversionReview"):
+		convertReview, ok := obj.(*v1beta1.ConversionReview)
+		if !ok {
+			msg := fmt.Sprintf("Expected v1beta1.ConversionReview but got: %T", obj)
+			klog.Errorf("%s", msg)
+			http.Error(w, msg, http.StatusBadRequest)
+			return
+		}
+		convertReview.Response = doConversionV1beta1(convertReview.Request, convert)
+		convertReview.Response.UID = convertReview.Request.UID
+		klog.V(2).Info(fmt.Sprintf("sending response: %v", convertReview.Response))
+
+		// reset the request, it is not needed in a response.
+		convertReview.Request = &v1beta1.ConversionRequest{}
+		responseObj = convertReview
```

Comment:
`ConversionReview` is GA since 1.16, IMHO we don't need to implement v1beta1 handling and register v1beta1   `conversionReviewVersions` in the CRD